docs/experimental/framebuffer_example.py

```diff
@@ -0,0 +1,246 @@
+
+
+
+scene.DebugOn()
+scene.GlobalWarningDisplayOn()
+
+manager = window.ShowManager(
+    scene,
+    "trees demo",
+    (1920,
+     1080),
+    reset_camera=True,
+    order_transparent=True)
+
+
+scale = np.array([[width, height, 0.0]])
+
+# Actor setup
+billboard = actor.billboard(np.array([[0.0, 0.0, 0.0]]), (1.0, 0.0, 0.0), scales=scale)
+
+frag_decl = shaders.compose_shader((frag_decl, kde_func))
+
+# Adding first the kde rendering shaders
+actor.shader_to_actor(billboard, "vertex", billboard_vert_impl, billboard_vert_decl)
+actor.shader_to_actor(billboard, "fragment", frag_impl, frag_decl)
+
+# billboard_tex = np.array([[0.0, 0.0], [0.0, 1.0], [1.0, 1.0], [1.0, 0.0]])
+billboard_tex = np.array([[-1.0, -1.0], [-1.0, 1.0], [1.0, 1.0], [1.0, -1.0]])
+actor.attribute_to_actor(billboard, billboard_tex, "in_tex")
+
+
+scene.GetRenderWindow().DebugOn()
+scene.GetRenderWindow().GlobalWarningDisplayOn()
+
+# Actor adding
+scene.add(billboard)
+
+
+
+
+
+# Points setup
+n_points = 10
+points = np.random.rand(10, 3)
+
+
+# FBO setup
+
+# Useful links
+# https://vtk.org/doc/nightly/html/classvtkOpenGLFramebufferObject.html#details
+# https://gitlab.kitware.com/vtk/vtk/-/blob/v9.2.0/Rendering/OpenGL2/Testing/Cxx/TestWindowBlits.cxx
+# https://vtk.org/doc/nightly/html/classvtkTextureObject.html
+# https://vtk.org/doc/nightly/html/classvtkOpenGLRenderWindow.html
+# https://learnopengl.com/Advanced-OpenGL/Framebuffers
+# https://github.com/JoaoDell/Comp-Grafica/blob/main/shadow3d.c
+# https://github.com/Kitware/VTK/blob/8f88edb91d2efea2d9cef1a1399d7a856c47f3be/Rendering/OpenGL2/vtkOpenGLFramebufferObject.cxx 
+# https://fury.gl/latest/auto_examples/04_demos/viz_fine_tuning_gl_context.html#sphx-glr-auto-examples-04-demos-viz-fine-tuning-gl-context-py
+
+
+
+
+FBO = vtk.vtkOpenGLFramebufferObject()
+
+manager.window.SetOffScreenRendering(True)
+manager.initialize() # sets everything for rendering
+
+FBO.DebugOn()
+FBO.GlobalWarningDisplayOn()
+
+FBO.SetContext(manager.window) # Sets the context for the FBO. 
+FBO.PopulateFramebuffer(width, height, True, 1, vtk.VTK_UNSIGNED_CHAR, False, 24, 0)
+
+
+
+
+# Checking FBO status
+print("FBO of index:", FBO.GetFBOIndex()) 
+print("Number of color attachments:", FBO.GetNumberOfColorAttachments())
+
+
+
+# RENDER TIME
+
+# Below, how to render things with a FBO according to VTK's website
+# FBO.SaveCurrentBindingsAndBuffers()
+# FBO.Bind()
+# FBO.ActivateBuffer(0)
+# scene.Clear()
+# FBO.Start(width, height)
+# # Render
+# FBO.RestorePreviousBindingsAndBuffers()
+
+shaders.shader_apply_effects(manager.window, billboard, window.gl_disable_depth)
+shaders.shader_apply_effects(manager.window, billboard, window.gl_set_normal_blending)
+
+
+FBO.SaveCurrentBindingsAndBuffers()
+FBO.Bind()
+FBO.ActivateBuffer(0)
+
+print("FBO Start:", FBO.Start(width, height))
+
+# Render every point existing inside loop
+for i in range(n_points):
+    shader_custom_uniforms(billboard, "fragment").SetUniform3f("point0", points[i, :])
```

Comment:
I don't believe this action is feasible as it would attempt to create new instances of 'point0' and 'res0' variables. This leads to an error. The error message I received :
```
2023-06-26 19:22:22.899 (   0.328s) [        B8DAB000]   vtkShaderProgram.cxx:453    ERR| vtkShaderProgram (0x558732a3cb20): 0:109(15): error: `res0' redeclared
0:110(15): error: `point0' redeclared
```

We have a old PR that deals with that 
https://github.com/fury-gl/fury/pull/424 maybe we need to look into this again @skoudoro  and @filipinascimento

Reply:
Interesting, I didn't know how VTK handled uniforms so I thought the `SetUniform` methods only passed uniform values to the shader. That seems like something worth checking out indeed 👍